.github/workflows/tests.yml

```diff
@@ -78,33 +78,40 @@ jobs:
     runs-on: ubuntu-latest
     steps:
 
-    - uses: actions/checkout@v3
-
-    - name: Setup Python
-      uses: actions/setup-python@v4
+      # https://stackoverflow.com/questions/74265821/get-modified-files-in-github-actions
+    - name: Checkout
+      uses: actions/checkout@v3
       with:
-        # Mypy can't run properly on 3.7 as it misses support for Literal types.
-        # FIXME once we drop support for 3.7, use the cache.
-        python-version: 3.8
+        fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
 
-    - name: Install dependencies
+    - name: Get changed files
+      id: complete-diff
+      run: |
+        if ${{ github.event_name == 'pull_request' }}; then
+            echo "changed_files=$(git diff --name-only -r HEAD^1 HEAD | xargs)" >> $GITHUB_OUTPUT
+        else
+            echo "changed_files=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }} | xargs)" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Filter changed files
```

Comment:
I would prefer using an action instead of custom code in these two steps, for example in the [tutorials PR workflow](https://github.com/deepset-ai/haystack-tutorials/blob/ee96e7e396e7dc4f1337126fddaee13d22b30290/.github/workflows/run_tutorials.yml#L37-L39) we use https://github.com/tj-actions/changed-files